src/libecalc/domain/stream_conditions.py

```diff
@@ -0,0 +1,114 @@
+"""
+Point in time stream conditions
+"""
+from __future__ import annotations
+
+import dataclasses
+import operator
+from dataclasses import dataclass
+from datetime import datetime
+from functools import reduce
+from typing import Dict, List
+
+from libecalc.common.string.string_utils import generate_id
+from libecalc.common.units import Unit
+
+
+@dataclass
+class Pressure:
+    value: float
+    unit: Unit
+
+    def __lt__(self, other: Pressure) -> bool:
+        if self.unit != other.unit:
+            raise ValueError("Unit mismatch")
```

Comment:
We should use domain specific exceptions, but they can perhaps extend ValueError in order to easily catch higher up?